src/util/child-component.js

```diff
@@ -103,3 +89,43 @@ export function createChildComponent (definition={}) {
 export function rejectNullElements(children) {
 	return _.reject(React.Children.toArray(children), (node) => (React.isValidElement(node) && node.type.isRenderNull));
 }
+
+export function filterElementsByType(children, elementTypes) {
+	return _.filter(React.Children.toArray(children), (node) => (React.isValidElement(node) && _.includes(elementTypes, node.type)));
+}
+
+export function getChildComponentPropsArray(props, childComponentType) {
+	const propName = _.get(childComponentType, 'childComponentDefinition.propName');
+	if (propName) {
+		const currentProps = _.chain(props)
+			.get(propName) // grab the prop we care about, e.g. "Child"
+			.thru(x => [x]) // wrap in an array
+			.flatten()
```

Comment:
I think it's because you can have multiple children or a single prop and we want to flatten either of those into a regular array.